wip/services/exec/store.go

```diff
@@ -117,10 +140,22 @@ func (p postgresStore) List(projectID *string, filterProvider *types.Provider, f
 	}
 	res := make([]types.Exec, len(execs))
 
-	for idx, exec := range execs {
-		exec := exec
-		res[idx] = dbExecToExec(exec)
+	for _, exec := range execs {
+		keyRefs, err := db.Q.ExecSSHKeysGetByExecID(ctx, exec.ID)
+		if err != nil {
+			return nil, err
+		}
+		keyIDs := db.MapStrings(keyRefs, func(key db.UnweaveExecSshKey) string {
+			return key.SshKeyID
+		})
+		keys, err := db.Q.SSHKeysGetByIDs(ctx, keyIDs)
+		if err != nil {
+			return []types.Exec{}, err
+		}
+
+		res = append(res, dbExecToExec(exec, dbSSHKeyToSSHKey(keys)))
```

Comment:
This is a bug. This appends to the end of an existing slice of `len(execs)` length. You should also avoid resizing the slice when know its length before hand.

Reply:
Thanks good catch, I missed a 0 here